src/main/java/edu/cornell/kfs/coa/batch/dataaccess/impl/ClosedLaborAccountsByDateRangeDaoJdbc.java

```diff
@@ -0,0 +1,74 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Locale;
+import java.util.Map;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.rice.core.framework.persistence.jdbc.dao.PlatformAwareDaoBaseJdbc;
+import org.springframework.jdbc.core.RowMapper;
+
+import edu.cornell.kfs.coa.batch.CuCoaBatchConstants;
+import edu.cornell.kfs.coa.batch.businessobject.LaborClosedAccount;
+import edu.cornell.kfs.coa.batch.dataaccess.ClosedLaborAccountsByDateRangeDao;
+
+
+public class ClosedLaborAccountsByDateRangeDaoJdbc extends PlatformAwareDaoBaseJdbc implements ClosedLaborAccountsByDateRangeDao {
+    private static final Logger LOG = LogManager.getLogger(ClosedLaborAccountsByDateRangeDaoJdbc.class);
+    
+    private static final String ACCT_FIN_COA_CD = "ACCT_FIN_COA_CD";
+    private static final String ACCT_ACCOUNT_NBR = "ACCT_ACCOUNT_NBR";
+    private static final String SUB_SUB_ACCT_NBR = "SUB_SUB_ACCT_NBR";
+    private static final String ACCT_ACCT_CLOSED_IND = "ACCT_ACCT_CLOSED_IND";
+    private static final String ACCTX_ACCT_CLOSED_DT = "ACCTX_ACCT_CLOSED_DT";
+    
+    @Override
+    public List<LaborClosedAccount> obtainLaborClosedAccountsDataFor(Map<String, Date> dateRange) {
+        List<LaborClosedAccount> closedLaborAccounts = findAllClosedLaborAccountsFor(dateRange);
+        return closedLaborAccounts;
+    }
+    
+    private List<LaborClosedAccount> findAllClosedLaborAccountsFor(Map<String, Date> dateRange) {
+        try {
+            RowMapper<LaborClosedAccount> mapRow = new RowMapper<LaborClosedAccount>() {
+                public LaborClosedAccount mapRow(ResultSet resultSet, int rowNumber) throws SQLException {
+                    LaborClosedAccount laborClosedAccountDataRow = new LaborClosedAccount();
+                    laborClosedAccountDataRow.setChart(resultSet.getString(ACCT_FIN_COA_CD));
+                    laborClosedAccountDataRow.setAccountNumber(resultSet.getString(ACCT_ACCOUNT_NBR));
+                    laborClosedAccountDataRow.setSubAccountNumber(resultSet.getString(SUB_SUB_ACCT_NBR));
+                    laborClosedAccountDataRow.setAccountClosedIndicator(resultSet.getString(ACCT_ACCT_CLOSED_IND));
+                    laborClosedAccountDataRow.setAccountClosedDate(resultSet.getDate(ACCTX_ACCT_CLOSED_DT));
+                    return laborClosedAccountDataRow;
+                }
+            };
+            return this.getJdbcTemplate().query(buildClosedLaborAccountsForDateRangeSQL(dateRange), mapRow);
+        } catch (Exception e) {
+            LOG.info("findAllClosedLaborAccountsFor Exception: " + e.getMessage());
+            return null;
+        }
+    }
+
+    private String buildClosedLaborAccountsForDateRangeSQL(Map<String, Date> dateRange) {
+        SimpleDateFormat formatter = new SimpleDateFormat("dd-MMM-yy", Locale.US);
```

Comment:
should we add "dd-MMM-yy" as a constant in CUKFSConstants, there are a number of date formats defined there.

Reply:
Added new constant.